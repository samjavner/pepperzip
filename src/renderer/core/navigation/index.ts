import * as actions from "./actions";
import {
    Area,
    CalendarPage,
    HomePage,
    moduleName,
    RecipesPage,
    ReferencePage,
    SettingsPage,
    ShoppingPage,
    ToolsPage,
} from "./model";
import { reducer } from "./reducer";
import selectors from "./selectors";

export { actions, moduleName, reducer, selectors };

export type Area = Area;
export type CalendarPage = CalendarPage;
export type HomePage = HomePage;
export type RecipesPage = RecipesPage;
export type ReferencePage = ReferencePage;
export type SettingsPage = SettingsPage;
export type ShoppingPage = ShoppingPage;
export type ToolsPage = ToolsPage;
