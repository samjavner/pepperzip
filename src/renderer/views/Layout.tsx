import {
    faArchive,
    faBook,
    faBox,
    faCalculator,
    faCalendar,
    faChartPie,
    faCog,
    faFileExport,
    faFileImport,
    faHeart,
    faHeartbeat,
    faHome,
    faInfoCircle,
    faSearch,
    faShoppingCart,
    faStopwatch,
    faTags,
} from "@fortawesome/free-solid-svg-icons";
import * as React from "react";
import {
    NavbarProps,
    NavigationLayout,
    TabProps,
} from "../components/navigation";
import { CommandProvider } from "../core/commandProvider";
import {
    CalendarPageName,
    HomePageName,
    Navigation,
    RecipesPageName,
    ReferencePageName,
    SettingsPageName,
    ShoppingPageName,
    ToolsPageName,
} from "../core/navigation";
import { RecipeBox } from "../core/recipes/recipeBox";
import { Import } from "../core/tools/import";
import { Statistics } from "../core/tools/statistics";
import CalendarPage from "./calendar/CalendarPage";
import HomePage from "./home/HomePage";
import FavoritesPage from "./recipes/FavoritesPage";
import LibraryPage from "./recipes/LibraryPage";
import RecipeBoxPage from "./recipes/RecipeBoxPage";
import SearchPage from "./recipes/SearchPage";
import TagsPage from "./recipes/TagsPage";
import DictionaryPage from "./reference/Dictionary";
import NutritionPage from "./reference/Nutrition";
import AboutPage from "./settings/AboutPage";
import SettingsPage from "./settings/SettingsPage";
import GroceryListsPage from "./shopping/GroceryListsPage";
import InventoryPage from "./shopping/InventoryPage";
import CalculatorPage from "./tools/CalculatorPage";
import ExportPage from "./tools/ExportPage";
import ImportPage from "./tools/ImportPage";
import StatisticsPage from "./tools/StatisticsPage";
import TimerPage from "./tools/TimerPage";

export interface LayoutProps {
    navigation: Navigation;
    recipeBox: RecipeBox;
    import: Import;
    statistics: Statistics;
    commandProvider: CommandProvider;
}

export const Layout: React.FunctionComponent<LayoutProps> = props => (
    <>
        <NavigationLayout
            navbar={getNavbar(props.navigation)}
            tabs={getTabs(props.navigation)}
            page={getPage(props)}
        />
        {(props.commandProvider.isInProgress || props.import.isImporting) && (
            <div className={"modal is-active"}>
                <div className="modal-background" />
                <div className="modal-content" />
                <button className="modal-close is-large" aria-label="close" />
            </div>
        )}
    </>
);

export default Layout;

// NAVBAR

function getNavbar({ activeArea, setActiveArea }: Navigation): NavbarProps {
    return {
        left: [
            {
                isActive: activeArea === "home",
                onClick: () => setActiveArea("home"),
                icon: faHome,
            },
            {
                isActive: activeArea === "recipes",
                onClick: () => setActiveArea("recipes"),
                text: "Recipes",
            },
            {
                isActive: activeArea === "calendar",
                onClick: () => setActiveArea("calendar"),
                text: "Calendar",
            },
            {
                isActive: activeArea === "shopping",
                onClick: () => setActiveArea("shopping"),
                text: "Shopping",
            },
            {
                isActive: activeArea === "reference",
                onClick: () => setActiveArea("reference"),
                text: "Reference",
            },
            {
                isActive: activeArea === "tools",
                onClick: () => setActiveArea("tools"),
                text: "Tools",
            },
        ],
        right: [
            {
                isActive: activeArea === "settings",
                onClick: () => setActiveArea("settings"),
                icon: faCog,
            },
        ],
    };
}

// TABS

function getTabs(navigation: Navigation): TabProps[] {
    switch (navigation.activeArea) {
        case "home":
            return getHomeTabs(
                navigation.activeHomePage,
                navigation.setActiveHomePage
            );
        case "recipes":
            return getRecipesTabs(
                navigation.activeRecipesPage,
                navigation.setActiveRecipesPage
            );
        case "calendar":
            return getCalendarTabs(
                navigation.activeCalendarPage,
                navigation.setActiveCalendarPage
            );
        case "shopping":
            return getShoppingTabs(
                navigation.activeShoppingPage,
                navigation.setActiveShoppingPage
            );
        case "reference":
            return getReferenceTabs(
                navigation.activeReferencePage,
                navigation.setActiveReferencePage
            );
        case "tools":
            return getToolsTabs(
                navigation.activeToolsPage,
                navigation.setActiveToolsPage
            );
        case "settings":
            return getSettingsTabs(
                navigation.activeSettingsPage,
                navigation.setActiveSettingsPage
            );
    }
}

function getHomeTabs(
    activePage: HomePageName,
    setActivePage: (page: HomePageName) => void
): TabProps[] {
    return [
        {
            isActive: activePage === "home",
            onClick: () => setActivePage("home"),
            icon: faHome,
            text: "Home",
        },
    ];
}

function getRecipesTabs(
    activePage: RecipesPageName,
    setActivePage: (page: RecipesPageName) => void
): TabProps[] {
    return [
        {
            isActive: activePage === "recipe_box",
            onClick: () => setActivePage("recipe_box"),
            icon: faArchive,
            text: "Recipe Box",
        },
        {
            isActive: activePage === "library",
            onClick: () => setActivePage("library"),
            icon: faBook,
            text: "Library",
        },
        {
            isActive: activePage === "favorites",
            onClick: () => setActivePage("favorites"),
            icon: faHeart,
            text: "Favorites",
        },
        {
            isActive: activePage === "tags",
            onClick: () => setActivePage("tags"),
            icon: faTags,
            text: "Tags",
        },
        {
            isActive: activePage === "search",
            onClick: () => setActivePage("search"),
            icon: faSearch,
            text: "Search",
        },
    ];
}

function getCalendarTabs(
    activePage: CalendarPageName,
    setActivePage: (page: CalendarPageName) => void
): TabProps[] {
    return [
        {
            isActive: activePage === "calendar",
            onClick: () => setActivePage("calendar"),
            icon: faCalendar,
            text: "Calendar",
        },
    ];
}

function getShoppingTabs(
    activePage: ShoppingPageName,
    setActivePage: (page: ShoppingPageName) => void
): TabProps[] {
    return [
        {
            isActive: activePage === "grocery_lists",
            onClick: () => setActivePage("grocery_lists"),
            icon: faShoppingCart,
            text: "Grocery Lists",
        },
        {
            isActive: activePage === "inventory",
            onClick: () => setActivePage("inventory"),
            icon: faBox,
            text: "Inventory",
        },
    ];
}

function getReferenceTabs(
    activePage: ReferencePageName,
    setActivePage: (page: ReferencePageName) => void
): TabProps[] {
    return [
        {
            isActive: activePage === "dictionary",
            onClick: () => setActivePage("dictionary"),
            icon: faBook,
            text: "Dictionary",
        },
        {
            isActive: activePage === "nutrition",
            onClick: () => setActivePage("nutrition"),
            icon: faHeartbeat,
            text: "Nutrition",
        },
    ];
}

function getToolsTabs(
    activePage: ToolsPageName,
    setActivePage: (page: ToolsPageName) => void
): TabProps[] {
    return [
        {
            isActive: activePage === "calculator",
            onClick: () => setActivePage("calculator"),
            icon: faCalculator,
            text: "Calculator",
        },
        {
            isActive: activePage === "timer",
            onClick: () => setActivePage("timer"),
            icon: faStopwatch,
            text: "Timer",
        },
        {
            isActive: activePage === "import",
            onClick: () => setActivePage("import"),
            icon: faFileImport,
            text: "Import",
        },
        {
            isActive: activePage === "export",
            onClick: () => setActivePage("export"),
            icon: faFileExport,
            text: "Export",
        },
        {
            isActive: activePage === "statistics",
            onClick: () => setActivePage("statistics"),
            icon: faChartPie,
            text: "Statistics",
        },
    ];
}

function getSettingsTabs(
    activePage: SettingsPageName,
    setActivePage: (page: SettingsPageName) => void
): TabProps[] {
    return [
        {
            isActive: activePage === "settings",
            onClick: () => setActivePage("settings"),
            icon: faCog,
            text: "Settings",
        },
        {
            isActive: activePage === "about",
            onClick: () => setActivePage("about"),
            icon: faInfoCircle,
            text: "About",
        },
    ];
}

// PAGE

function getPage(props: LayoutProps): React.ReactElement<any> {
    switch (props.navigation.activeArea) {
        case "home":
            return getHomePage(props);
        case "recipes":
            return getRecipesPage(props);
        case "calendar":
            return getCalendarPage(props);
        case "shopping":
            return getShoppingPage(props);
        case "reference":
            return getReferencePage(props);
        case "tools":
            return getToolsPage(props);
        case "settings":
            return getSettingsPage(props);
    }
}

function getHomePage(props: LayoutProps): React.ReactElement<any> {
    switch (props.navigation.activeHomePage) {
        case "home":
            return <HomePage />;
    }
}

function getRecipesPage(props: LayoutProps): React.ReactElement<any> {
    switch (props.navigation.activeRecipesPage) {
        case "recipe_box":
            return <RecipeBoxPage recipeBox={props.recipeBox} />;
        case "library":
            return <LibraryPage />;
        case "favorites":
            return <FavoritesPage />;
        case "tags":
            return <TagsPage />;
        case "search":
            return <SearchPage />;
    }
}

function getCalendarPage(props: LayoutProps): React.ReactElement<any> {
    switch (props.navigation.activeCalendarPage) {
        case "calendar":
            return <CalendarPage />;
    }
}

function getShoppingPage(props: LayoutProps): React.ReactElement<any> {
    switch (props.navigation.activeShoppingPage) {
        case "grocery_lists":
            return <GroceryListsPage />;
        case "inventory":
            return <InventoryPage />;
    }
}

function getReferencePage(props: LayoutProps): React.ReactElement<any> {
    switch (props.navigation.activeReferencePage) {
        case "dictionary":
            return <DictionaryPage />;
        case "nutrition":
            return <NutritionPage />;
    }
}

function getToolsPage(props: LayoutProps): React.ReactElement<any> {
    switch (props.navigation.activeToolsPage) {
        case "calculator":
            return <CalculatorPage />;
        case "timer":
            return <TimerPage />;
        case "import":
            return <ImportPage {...props.import} />;
        case "export":
            return <ExportPage />;
        case "statistics":
            return <StatisticsPage {...props.statistics} />;
    }
}

function getSettingsPage(props: LayoutProps): React.ReactElement<any> {
    switch (props.navigation.activeSettingsPage) {
        case "settings":
            return <SettingsPage />;
        case "about":
            return <AboutPage />;
    }
}
