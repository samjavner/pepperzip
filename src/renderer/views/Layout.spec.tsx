import * as React from "react";
import * as renderer from "react-test-renderer";
import * as Model from "../core/navigation";
import { Layout, LayoutProps } from "./Layout";

describe("Layout", () => {
    let props: LayoutProps;

    beforeEach(() => {
        props = {
            navigation: {
                ...Model.init,
                setActiveArea: () => ({}),
                setActiveHomePage: () => ({}),
                setActiveRecipesPage: () => ({}),
                setActiveCalendarPage: () => ({}),
                setActiveShoppingPage: () => ({}),
                setActiveReferencePage: () => ({}),
                setActiveToolsPage: () => ({}),
                setActiveSettingsPage: () => ({}),
            },
            commandProvider: {
                isInProgress: false,
            } as any,
            recipeBox: {} as any,
            import: {
                isImporting: false,
            } as any,
            statistics: {} as any,
        };
    });

    it("renders the navbar, tabs, and active page", () => {
        const component = renderer.create(<Layout {...props} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("loading overlay", () => {
        it("should be there when commandProvider is executing", () => {
            props.commandProvider.isInProgress = true;
            const component = renderer.create(<Layout {...props} />);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });

        it("should be there when importing recipes", () => {
            props.import.isImporting = true;
            const component = renderer.create(<Layout {...props} />);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
