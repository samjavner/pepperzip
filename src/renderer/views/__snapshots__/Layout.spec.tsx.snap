// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout loading overlay should be there when commandProvider is executing 1`] = `
Array [
  <div
    style={
      Object {
        "display": "grid",
        "gridTemplateRows": "auto 1fr",
        "height": "100vh",
        "width": "100vw",
      }
    }
  >
    <div
      style={
        Object {
          "gridRow": 1,
        }
      }
    >
      <nav
        className="navbar is-primary"
      >
        <div
          className="navbar-brand"
        >
          <a
            className="navbar-item is-active"
            href="#"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-home fa-w-18 fa-fw "
              data-icon="home"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Recipes
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Calendar
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Shopping
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Reference
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Tools
          </a>
        </div>
        <div
          className="navbar-menu"
        >
          <div
            className="navbar-end"
          >
            <a
              className="navbar-item"
              href="#"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-cog fa-w-16 fa-fw "
                data-icon="cog"
                data-prefix="fas"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
        </div>
      </nav>
    </div>
    <div
      style={
        Object {
          "gridRow": 2,
          "overflow": "hidden",
        }
      }
    >
      <div
        style={
          Object {
            "display": "grid",
            "gridTemplateRows": "auto 1fr",
            "height": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "gridRow": 1,
            }
          }
        >
          <div
            className="tabs is-centered"
            style={
              Object {
                "marginBottom": "0px",
              }
            }
          >
            <ul>
              <li
                className="is-active"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <span
                    className="icon is-small"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-home fa-w-18 "
                      data-icon="home"
                      data-prefix="fas"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </span>
                  Home
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          style={
            Object {
              "gridRow": 2,
              "overflow": "hidden",
            }
          }
        >
          <div>
            Home
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="modal is-active"
  >
    <div
      className="modal-background"
    />
    <div
      className="modal-content"
    />
    <button
      aria-label="close"
      className="modal-close is-large"
    />
  </div>,
]
`;

exports[`Layout loading overlay should be there when importing recipes 1`] = `
Array [
  <div
    style={
      Object {
        "display": "grid",
        "gridTemplateRows": "auto 1fr",
        "height": "100vh",
        "width": "100vw",
      }
    }
  >
    <div
      style={
        Object {
          "gridRow": 1,
        }
      }
    >
      <nav
        className="navbar is-primary"
      >
        <div
          className="navbar-brand"
        >
          <a
            className="navbar-item is-active"
            href="#"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-home fa-w-18 fa-fw "
              data-icon="home"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Recipes
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Calendar
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Shopping
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Reference
          </a>
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            Tools
          </a>
        </div>
        <div
          className="navbar-menu"
        >
          <div
            className="navbar-end"
          >
            <a
              className="navbar-item"
              href="#"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-cog fa-w-16 fa-fw "
                data-icon="cog"
                data-prefix="fas"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
        </div>
      </nav>
    </div>
    <div
      style={
        Object {
          "gridRow": 2,
          "overflow": "hidden",
        }
      }
    >
      <div
        style={
          Object {
            "display": "grid",
            "gridTemplateRows": "auto 1fr",
            "height": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "gridRow": 1,
            }
          }
        >
          <div
            className="tabs is-centered"
            style={
              Object {
                "marginBottom": "0px",
              }
            }
          >
            <ul>
              <li
                className="is-active"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <span
                    className="icon is-small"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-home fa-w-18 "
                      data-icon="home"
                      data-prefix="fas"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </span>
                  Home
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          style={
            Object {
              "gridRow": 2,
              "overflow": "hidden",
            }
          }
        >
          <div>
            Home
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="modal is-active"
  >
    <div
      className="modal-background"
    />
    <div
      className="modal-content"
    />
    <button
      aria-label="close"
      className="modal-close is-large"
    />
  </div>,
]
`;

exports[`Layout renders the navbar, tabs, and active page 1`] = `
<div
  style={
    Object {
      "display": "grid",
      "gridTemplateRows": "auto 1fr",
      "height": "100vh",
      "width": "100vw",
    }
  }
>
  <div
    style={
      Object {
        "gridRow": 1,
      }
    }
  >
    <nav
      className="navbar is-primary"
    >
      <div
        className="navbar-brand"
      >
        <a
          className="navbar-item is-active"
          href="#"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-home fa-w-18 fa-fw "
            data-icon="home"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <a
          className="navbar-item"
          href="#"
          onClick={[Function]}
        >
          Recipes
        </a>
        <a
          className="navbar-item"
          href="#"
          onClick={[Function]}
        >
          Calendar
        </a>
        <a
          className="navbar-item"
          href="#"
          onClick={[Function]}
        >
          Shopping
        </a>
        <a
          className="navbar-item"
          href="#"
          onClick={[Function]}
        >
          Reference
        </a>
        <a
          className="navbar-item"
          href="#"
          onClick={[Function]}
        >
          Tools
        </a>
      </div>
      <div
        className="navbar-menu"
      >
        <div
          className="navbar-end"
        >
          <a
            className="navbar-item"
            href="#"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-cog fa-w-16 fa-fw "
              data-icon="cog"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </div>
      </div>
    </nav>
  </div>
  <div
    style={
      Object {
        "gridRow": 2,
        "overflow": "hidden",
      }
    }
  >
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateRows": "auto 1fr",
          "height": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "gridRow": 1,
          }
        }
      >
        <div
          className="tabs is-centered"
          style={
            Object {
              "marginBottom": "0px",
            }
          }
        >
          <ul>
            <li
              className="is-active"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <span
                  className="icon is-small"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-home fa-w-18 "
                    data-icon="home"
                    data-prefix="fas"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </span>
                Home
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        style={
          Object {
            "gridRow": 2,
            "overflow": "hidden",
          }
        }
      >
        <div>
          Home
        </div>
      </div>
    </div>
  </div>
</div>
`;
